File: project-04-dynamic-search/code-java/src/main/java/com/example/demo/dto/ProductSearchRequestDto.java
package com.example.demo.dto;

import java.math.BigDecimal;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ProductSearchRequestDto {
    private String keyword;
    private String category;
    private BigDecimal minPrice;
    private BigDecimal maxPrice;
    private String stockStatus;
    private String status;
    private String manufacturer;
}

File: project-04-dynamic-search/code-kotlin/src/main/kotlin/com/example/demo/dto/ProductSearchRequestDto.kt
package com.example.demo.dto

import java.math.BigDecimal

data class ProductSearchRequestDto(
    val keyword: String? = null,
    val category: String? = null,
    val minPrice: BigDecimal? = null,
    val maxPrice: BigDecimal? = null,
    val stockStatus: String? = null,
    val status: String? = null,
    val manufacturer: String? = null
)
